pilot/pkg/serviceregistry/kube/controller/namespacecontroller.go

```diff
@@ -135,6 +132,39 @@ func (nc *NamespaceController) Run(stopCh <-chan struct{}) {
 	}
 	log.Infof("Namespace controller started")
 	go nc.queue.Run(stopCh)
+	go nc.startCaBundleWatcher(stopCh)
+}
+
+// startCaBundleWatcher listens for updates to the CA bundle and update cm in each namespace
+func (nc *NamespaceController) startCaBundleWatcher(stop <-chan struct{}) {
+	task := func() error {
+		namespaceList, err := nc.namespaceLister.List(labels.Everything())
+		if err != nil {
+			return err
+		}
+
+		for _, ns := range namespaceList {
+			cm, err := nc.configmapLister.ConfigMaps(ns.Name).Get(CACertNamespaceConfigMap)
+			if err != nil {
+				continue
+			}
+			err = nc.configMapChange(cm)
```

Comment:
Seems like we should push them on the queue. Although your issue about queue improvement will make it better

Reply:
it is in the `task` cb, which is pushed to the queue

Reply:
Yeah but we are pushing one task which writes 1000s of events, instead of pushing 1000 tasks that write 1 event.

If any 1 of the 1000 namespaces fails we would retry the whole thing

Reply:
Agree with John's comment - think its a good idea to have each namespace update as a separate task. Also, does it make sense to change the task definition to actually include the caBundle as an argument? Wondering what happens if two changes to the caBundle happen quickly, so that one set of namespaces get updated with update1 and the other with update2? Eventually, all of them will reconcile to the latest, but could this be an issue?

Reply:
IC,  i am ok with this